gnomad/utils/constraint.py

```diff
@@ -125,3 +135,246 @@ def collapse_strand(
+    and will be used to build plateau models. Sites below this cutoff have low coverage and will be used to build
+    coverage models. Defaults to `COVERAGE_CUTOFF`.
+    :return: Coverage model and plateau models.
+    """
+    # Filter to sites with coverage above `cov_cutoff`
+    all_high_coverage_ht = coverage_ht.filter(coverage_ht.exome_coverage >= cov_cutoff)
+    agg_expr = {
+        "observed_variants": hl.agg.sum(all_high_coverage_ht.observed_variants),
+        "possible_variants": hl.agg.sum(all_high_coverage_ht.possible_variants),
+    }
+    for pop in pops:
+        agg_expr[f"observed_{pop}"] = hl.agg.array_sum(
+            all_high_coverage_ht[f"downsampling_counts_{pop}"]
+        )
+    # Generate a Table with all necessary annotations (x and y listed above) for the plateau models
+    high_coverage_ht = all_high_coverage_ht.group_by(*keys).aggregate(**agg_expr)
+    high_coverage_ht = annotate_mutation_type(high_coverage_ht)
+    # Build plateau models
+    plateau_models_agg_expr = build_plateau_models(
+        high_coverage_ht,
+        cpg=high_coverage_ht.cpg,
+        mu_snp=high_coverage_ht.mu_snp,
+        observed_variants=high_coverage_ht.observed_variants,
+        possible_variants=high_coverage_ht.possible_variants,
+        weighted=weighted,
+        pops=pops,
+    )
+    plateau_models = high_coverage_ht.aggregate(hl.struct(**plateau_models_agg_expr))
+
+    # Filter to sites with coverage below `cov_cutoff and larger than 0
+    all_low_coverage_ht = coverage_ht.filter(
+        (coverage_ht.exome_coverage < cov_cutoff) & (coverage_ht.exome_coverage > 0)
+    )
+    # Metric that represents the relative mutability of the exome calculated on high coverage sites
+    # This metric is used as scaling factor when building the coverage model
+    high_coverage_scale_factor = all_high_coverage_ht.aggregate(
+        hl.agg.sum(all_high_coverage_ht.observed_variants)
+        / hl.agg.sum(
+            all_high_coverage_ht.possible_variants * all_high_coverage_ht.mu_snp
+        )
+    )
+    # Generate a Table with all necessary annotations (x and y listed above) for the coverage model
+    low_coverage_ht = all_low_coverage_ht.group_by(
+        log_coverage=hl.log10(all_low_coverage_ht.exome_coverage)
+    ).aggregate(
+        low_coverage_obs_exp=hl.agg.sum(all_low_coverage_ht.observed_variants)
+        / (
+            high_coverage_scale_factor
+            * hl.agg.sum(
+                all_low_coverage_ht.possible_variants * all_low_coverage_ht.mu_snp
+            )
+        )
+    )
+    # Build the coverage model
+    # TODO: consider weighting here as well
+    coverage_model_expr = build_coverage_model(
+        low_coverage_obs_exp=low_coverage_ht.low_coverage_obs_exp,
+        log_coverage=low_coverage_ht.log_coverage,
+    )
+    coverage_model = tuple(low_coverage_ht.aggregate(coverage_model_expr).beta)
```

Comment:
Tbh I don't really like the idea to take the `ht` param out of the function. The function is not a complete function without this line of code. If the users only use the function without this line of code, they get nothing.

Reply:
Sorry I didn't say I idea clearly. I felt if the users call `build_coverage_model()` they won't get a complete coverage model, because the coverage model haven't been aggregated on this line. I just worry that the users may be confused on the uncompleted model.

Reply:
oh, I see! I don't think this is an issue given that many of our functions are similar conceptually (they return HailExpressions that aren't meaningful until actually evaluated elsewhere)